### PR TITLE
Larger staging listing batch size for commit

### PR DESCRIPTION
**Pull request by guy-har**

### CODE REVIEW

**Review by nopcoder — CHANGES_REQUESTED** [review]
If all is internal, I prefer to pass the constant with the value. the iterator can handle 0 and -1 in case of user input.

File: pkg/graveler/staging/iterator.go [hunk 1]
```diff
@@ -25,10 +23,19 @@ type Iterator struct {
...
 
-func NewStagingIterator(ctx context.Context, db db.Database, log logging.Logger, st graveler.StagingToken) *Iterator {
-	return &Iterator{ctx: ctx, st: st, dbHasNext: true, initPhase: true, db: db, log: log, nextFrom: make([]byte, 0)}
+// NewStagingIterator initiates the staging iterator with a batchSize
+func NewStagingIterator(ctx context.Context, db db.Database, log logging.Logger, st graveler.StagingToken, bachSize int) *Iterator {
+	bs := bachSize
+	if bs <= 0 {
+		bs = graveler.ListingDefaultBatchSize
+	}
+	if bs > graveler.ListingMaxBatchSize {
```
Comment (nopcoder) [inline pkg/graveler/staging/iterator.go:35]: else if

**Review by nopcoder — APPROVED** [review]
LGTM

File: pkg/graveler/staging/iterator.go [hunk 2]
```diff
@@ -25,10 +23,18 @@ type Iterator struct {
 	dbHasNext   bool
 	buffer      []*graveler.ValueRecord
 	nextFrom    graveler.Key
+	batchSize   int
 }
 
-func NewStagingIterator(ctx context.Context, db db.Database, log logging.Logger, st graveler.StagingToken) *Iterator {
-	return &Iterator{ctx: ctx, st: st, dbHasNext: true, initPhase: true, db: db, log: log, nextFrom: make([]byte, 0)}
+// NewStagingIterator initiates the staging iterator with a batchSize
+func NewStagingIterator(ctx context.Context, db db.Database, log logging.Logger, st graveler.StagingToken, bachSize int) *Iterator {
```
Comment (itaiad200) [inline pkg/graveler/staging/iterator.go:30]:
```suggestion
func NewStagingIterator(ctx context.Context, db db.Database, log logging.Logger, st graveler.StagingToken, batchSize int) *Iterator {
```

Comment (guy-har) [reply pkg/graveler/staging/iterator.go:30]: Thanks!

**Pull request merged**